src/Driver/AbstractMySQLDriver.php

```diff
@@ -58,7 +59,17 @@ public function getDatabasePlatform(ServerVersionProvider $versionProvider): Abs
             return new MySQL84Platform();
         }
 
+        if (version_compare($version, '8.0.13', '>=')) {
+            return new MySQL8013Platform();
+        }
+
         if (version_compare($version, '8.0.0', '>=')) {
+            Deprecation::trigger(
+                'doctrine/dbal',
+                'https://github.com/doctrine/dbal/pull/6343',
+                'Support for MySQL <= 8.0.13 is deprecated and will be removed in DBAL 5',
+            );
```

Comment:
If you move the deprection trigger above the `if` statement, you can remove the other deprecation further below. Please also add a note to our upgrade guide about the new deprecation.

Reply:
Please look it closer, I think I'm made it right. There is a mariadb branch, mysql 8.0 and higher and default branch for lower(5.7 is still supported, despite of deprecation).

Reply:
After your change, we have a deprecation message for MySQL < 8.0.13 and another one for MySQL < 8.0. The second one is redundant.